webapp/src/ts/services/performance.service.ts

```diff
@@ -0,0 +1,92 @@
+import { Inject, Injectable } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+
+import { TelemetryService } from '@mm-services/telemetry.service';
+import { AuthService } from '@mm-services/auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PerformanceService {
+  private trackPerformance = false;
+  private readonly CAN_TRACK_PERFORMANCE = 'track_performance';
+  private readonly PERFORMANCE_PREFIX = 'perf:';
+  private readonly APDEX_SUBFIX = ':apdex';
+  private readonly APDEX_SATISFIED = ':satisfied';
+  private readonly APDEX_TOLERATING = ':tolerable';
+  private readonly APDEX_FRUSTRATED = ':frustrated';
+  private readonly APDEX_AGGREGATE = ':aggregate';
+  private readonly APDEX_T = 3 * 1000;
+  private readonly APDEX_TOLERANCE = 4; // 4xT
+
+  constructor(
+    private telemetryService: TelemetryService,
+    private authService: AuthService,
+    @Inject(DOCUMENT) private document: Document,
+  ) {
+    this.authService
+      .has(this.CAN_TRACK_PERFORMANCE)
+      .then(result => this.trackPerformance = result);
+  }
+
+  track(name: string) {
+    if (!this.trackPerformance || !name || !this.document?.defaultView) {
+      return;
+    }
+
+    const startTime = this.document.defaultView.performance.now();
+    return {
+      setName: newName => name = newName,
+      stop: (recordApdex = false, prefix = true) => this.recordPerformance(name, startTime, recordApdex, prefix),
```

Comment:
Having multiple booleans in an API is annoying because when you're reading the code you have no idea what `stop(true, false)` means and have to dig down deeper.

Consider changing this to something like...

```suggestion
      stop: ({ recordApdex = false, prefix = true }) => this.recordPerformance(name, startTime, recordApdex, prefix),
```